src/components/TodoFooter.tsx

```diff
@@ -0,0 +1,67 @@
+import classNames from 'classnames';
+import { useTodo } from '../providers/TodoProvider';
+import { Filter } from '../types/Filter';
+
+export const TodoFooter = () => {
+  const {
+    todos, activeFilter, setActiveFilter, todosLeft,
+  } = useTodo();
+
+  const handleClick = (filter: Filter) => () => {
+    setActiveFilter(filter);
+  };
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {`${todosLeft} items left`}
+      </span>
+
+      {/* Active filter should have a 'selected' class */}
+      <nav className="filter" data-cy="Filter">
+        <a
+          href="#/"
+          className={classNames('filter__link', {
+            selected: activeFilter === 'all',
+          })}
+          data-cy="FilterLinkAll"
+          onClick={handleClick('all')}
+        >
+          All
+        </a>
```

Comment:
You can move the filter to another component and create an array to map filter values. In this scenario, if a value is repeated three times, you can express this more clearly

For Example
`const FILTER_VALUES = ['All', 'Active', 'Completed'];
        {FILTER_VALUES.map(filter => (
          <Filter filter={filter} key={filter} />
        ))}
`

Reply:
That is a great idea.